compel/src/lib/ptrace.c

```diff
@@ -21,6 +23,68 @@
 
 #include "log.h"
 
+int ptrace_get_sud(pid_t tid, sud_config_t *cfg)
+{
+	if (ptrace(PTRACE_GET_SYSCALL_USER_DISPATCH_CONFIG, tid, (void *)sizeof(*cfg), cfg)) {
+		pr_perror("getting SUD config failed");
+		return -1;
+	}
+	return 0;
+}
+
+int ptrace_set_sud(pid_t tid, sud_config_t *set)
+{
+	if (ptrace(PTRACE_SET_SYSCALL_USER_DISPATCH_CONFIG, tid, (void *)sizeof(*set), set)) {
+		pr_perror("setting SUD config failed");
+		return -1;
+	}
+	return 0;
+}
+
+int ptrace_suspend_sud(pid_t pid)
+{
+	char task[32];
+	int tid;
+	sud_config_t disable;
+	struct dirent *de;
+	DIR *dp;
+
+	/* Setup SUD-disable struct */
+	memset(&disable, 0, sizeof(disable));
+	disable.mode = PR_SYS_DISPATCH_OFF;
```

Comment:
```
#10 2.352 compel/src/lib/ptrace.c:54:17: error: ‘PR_SYS_DISPATCH_OFF’ undeclared (first use in this function)
#10 2.352    54 |  disable.mode = PR_SYS_DISPATCH_OFF;
#10 2.352       |                 ^~~~~~~~~~~~~~~~~~~
```

Reply:
This is probably a case of kernel headers not always being defined. I'll go back and explicitly add the #ifdefs to sud.h.